### PR TITLE
Fix comment removing issue of unit types in struct fields

### LINKED ISSUE #4749: Rustfmt eats comments inside unit type in struct/enum definitions
**Describe the bug**

Rustfmt eats comments inside the unit type `()` in struct and enum definitions. I sometimes find it useful to put comments there when I am not sure what type something will be, but I know I will need it later.

**To Reproduce**

Run Rustfmt on the following code:

```
struct S {
    x: (/* type */),
}

struct S2((/* type */));

enum E {
    Variant1 {
        x: (/* type */),
    },
    Variant2((/* type */)),
}
```

[Playground link](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=61030a9a75a263e454885e9ffb673386)

**Expected behavior**

All four comments should remain intact.

**Meta**

- rustfmt version: 1.4.36-nightly (2021-02-07 7de6968)
- From where did you install rustfmt?: Rust Playground
- How do you run rustfmt: Tools -> Rustfmt

### PR DESCRIPTION
**Pull request by whizsid**
Fixes #4749

### CODE REVIEW

File: src/formatting/overflow.rs [hunk 1]
```diff
@@ -588,6 +589,13 @@ impl<'a> Context<'a> {
 
     fn rewrite_items(&self) -> Option<(bool, String)> {
         let span = self.items_span();
+        if self.items.len() == 0 && contains_comment(self.context.snippet(span)) {
```
Comment (calebcartwright) [inline src/formatting/overflow.rs:592]: nit:

```suggestion
        if self.items.is_empty() && contains_comment(self.context.snippet(span)) {
``

File: src/formatting/overflow.rs [hunk 2]
```diff
@@ -588,6 +589,13 @@ impl<'a> Context<'a> {
...
+            let hi = self.context.snippet_provider.span_before(span, self.suffix);
+            let cmnt =
+                rewrite_missing_comment(mk_sp(span.lo(), hi), self.nested_shape, self.context)?;
```
Comment (calebcartwright) [inline src/formatting/overflow.rs:595]: nit:
```suggestion
                rewrite_missing_comment(span.to(hi), self.nested_shape, self.context)?;
```

**Review by calebcartwright — COMMENTED** [review]
Thanks! Fix looks good, few minor cleanup nits and then ready to merge

File: src/formatting/overflow.rs (hunk 2 above)
Comment (whizsid) [reply src/formatting/overflow.rs]: As per the [doc](https://docs.rs/rustc-ap-rustc_span/712.0.0/rustc_ap_rustc_span/struct.Span.html#method.to) `to` method is taking a `Span` as the first argument. But `hi` is a `BytePos`. So I used the [with_hi](https://docs.rs/rustc-ap-rustc_span/712.0.0/rustc_ap_rustc_span/struct.Span.html#method.with_hi) method.

**Comment by calebcartwright** [comment]
> As per the doc to method is taking a Span as the first argument. But hi is a BytePos. So I used the with_hi method.

Indeed! (and this is why it's easier to code in an editor and not the GitHub browser UI :laughing:)

Good you grab the latest nightly toolchain and updates from master, and then rebase when you get a chance?

**Comment by calebcartwright** [comment]
Thanks for the updates! For future reference, please be sure to do a rebase instead of adding branch merge commits, as we want to avoid adding merge commits to the index. Will go ahead and squash the PR on merging it so no worries, but a standard rebase will definitely be helpful in any future cases.

**Pull request merged**